File: 129.cpp
#include<iostream>
#include<vector>
#include "Tree.h"
using namespace std;

class Solution {
public:
	int sumNumbers(TreeNode* root) {
		int sum = 0;
		get(root,sum,0);
		return sum;
	}
	void get(TreeNode* root,int &sum,int num) {
		if (!root)
			return;
		num = num * 10 + root->val;
		if (!root->left && !root->right)
			sum += num;
		else {
			get(root->left, sum, num);
			get(root->right, sum, num);
		}
	}
};
int main() {
	TreeEnv t;
	vector<int> list = {1,2,3};
	TreeNode* root = t.initialTree(list);
	Solution s;
	cout << s.sumNumbers(root) << endl;

	return 0;

}

File: Python3/129.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        def getSum(root, val):
            if not root:
                return 0
            new_val = val*10 + root.val
            if not root.left and not root.right:
                return new_val
            return getSum(root.left, new_val) + getSum(root.right, new_val)
        return getSum(root, 0) 
